lib/dom.js

```diff
@@ -1060,6 +1068,167 @@ Node.prototype = {
 	 * @type {string | null}
 	 */
 	localName: null,
+	/**
+	 * The URI is currently always `about:blank` since that's what happens when you create a
+	 * document from scratch.
+	 * This could be improved by making it possible to add a baseURI to the document when creating
+	 * one so it's childNodes just use it's URI instead.
+	 *
+	 * @type {string}
+	 */
+	baseURI: 'about:blank',
+	/**
+	 * Is true if this node is part of a document.
+	 *
+	 * @type {boolean}
+	 */
+	get isConnected() {
+		var rootNode = this.getRootNode();
+		if (rootNode && rootNode.nodeType === rootNode.DOCUMENT_NODE) {
+			return true;
+		}
+
+		return false;
+	},
+	/**
+	 * Checks whether or not `other` is an inclusive descendant of this node.
+	 *
+	 * @param {Node} [other]
+	 * The node to check.
+	 * @returns {boolean}
+	 * True if `other` is an inclusive descendant of this node; false otherwise.
+	 * @see https://dom.spec.whatwg.org/#dom-node-contains
+	 */
+	contains: function (other) {
+		if (!other) {
+			return false;
+		}
+
+		do {
+			if (this === other) {
+				return true;
+			}
+		} while ((other = other.parentNode));
+
+		return false;
+	},
+	/**
+	 * @typedef GetRootNodeOptions
+	 * @property {boolean} [composed=false]
+	 */
+	/**
+	 * Searches for the root node of this node.
+	 *
+	 * Difference to the specs:
+	 * options.compose doesn't change this behaviour.
+	 *
+	 * @param {GetRootNodeOptions} [options]
+	 * @returns {Node}
+	 * Root node.
+	 * @see https://dom.spec.whatwg.org/#dom-node-getrootnode
+	 */
+	getRootNode: function (options) {
+		if (options && options.composed) {
+			// should return shadow root here (not supported)
```

Comment:
Question: Should we rather throw `new Error('not implemented: composed === true');`, instead of returning the wrong value? Or even go the extra mile of implementing that case?

Reply:
Just looked it up, since we do not have/support `ShadowRoot`s so far that would be a bit to much.
and also even when `composed === true`, it would only return the shadow root if the root is a shadow root, otherwise it would return the normal root. So throwing doesn't make sense.

I will tweak the comment accordingly.